packages/cli/src/transfer-stable-base.ts

```diff
@@ -36,35 +35,60 @@ export abstract class TransferStableBase extends BaseCommand {
     } catch {
       failWith(`The ${this._stableCurrency} token was not deployed yet`)
     }
+
     // If gasCurrency is not set, use the transferring token
-    if (!res.flags.gasCurrency) {
-      await kit.setFeeCurrency(stableTokenInfos[this._stableCurrency].contract)
+    if (res.flags.gasCurrency) {
+      kit.connection.defaultFeeCurrency = res.flags.gasCurrency
+    } else {
+      kit.connection.defaultFeeCurrency = (
+        await kit.contracts.getStableToken(this._stableCurrency)
+      ).address
     }
 
     const tx = res.flags.comment
       ? stableToken.transferWithComment(to, value.toFixed(), res.flags.comment)
       : stableToken.transfer(to, value.toFixed())
 
+    let gas: number
+    let gasPrice: string
+    let gasBalance: BigNumber
+    let valueBalance: BigNumber
     await newCheckBuilder(this)
       .hasEnoughStable(from, value, this._stableCurrency)
       .isNotSanctioned(from)
       .isNotSanctioned(to)
-      .addConditionalCheck(
-        `Account can afford transfer and gas paid in ${this._stableCurrency}`,
-        kit.connection.defaultFeeCurrency === stableToken.address,
+      .addCheck(
+        `Account can afford transfer and gas paid in ${
+          res.flags.gasCurrency || this._stableCurrency
+        }`,
         async () => {
-          const [gas, gasPrice, balance] = await Promise.all([
-            tx.txo.estimateGas({ feeCurrency: stableToken.address }),
-            kit.connection.gasPrice(stableToken.address),
-            stableToken.balanceOf(from),
+          ;[gas, gasPrice, gasBalance, valueBalance] = await Promise.all([
+            tx.txo.estimateGas({ feeCurrency: kit.connection.defaultFeeCurrency }),
+            kit.connection.gasPrice(kit.connection.defaultFeeCurrency),
+            kit.contracts
+              .getErc20(kit.connection.defaultFeeCurrency!)
+              .then((erc20) => erc20.balanceOf(from)),
+            kit.contracts
+              .getStableToken(this._stableCurrency!)
+              .then((token) => token.balanceOf(from)),
           ])
           const gasValue = new BigNumber(gas).times(gasPrice as string)
-          return balance.gte(value.plus(gasValue))
+          if (res.flags.gasCurrency) {
+            return gasBalance.gte(gasValue) && valueBalance.gte(value)
+          }
+          return valueBalance.gte(value.plus(gasValue))
         },
         `Cannot afford transfer with ${this._stableCurrency} gasCurrency; try reducing value slightly or using gasCurrency=CELO`
       )
       .runChecks()
 
-    await displaySendTx(res.flags.comment ? 'transferWithComment' : 'transfer', tx)
+    await displaySendTx(res.flags.comment ? 'transferWithComment' : 'transfer', tx, {
+      feeCurrency: kit.connection.defaultFeeCurrency,
+      // NOTE(nico): passing this makes the tx eip1559 rather than legacy-celo
```

Comment:
hmm celo-legacy requires explicitly passing gasPrice. that must be happening further down

Reply:
na celo/connect will set the `maxFeesPerGas` not the `gasPrice`. 

the ledger fix happily only happens in the ledger wallet 

https://github.com/celo-org/developer-tooling/blob/aaronmgdr/beta-simple/packages/sdk/wallets/wallet-ledger/src/ledger-wallet.ts#L71-L72

Reply:
so i dont think "// NOTE(nico): passing this makes the tx eip1559 rather than legacy-celo" is accurate. 

 because there is feeCurrency it will be a CIP64 tx. and whether the maxFeePerGas and maxPriorityFeePerGas are added or not here they will be added automatically in the tx params normalizer

Reply:
changed the behaviour as per our slack discussion, will recap in the changelogs